src/main.cpp

```diff
@@ -6615,16 +6654,13 @@ static class CMainCleanup
 // Set default values of new CMutableTransaction based on consensus rules at given height.
 CMutableTransaction CreateNewContextualCMutableTransaction(const Consensus::Params& consensusParams, int nHeight)
 {
-    return CreateNewContextualCMutableTransaction(consensusParams, nHeight, expiryDelta);
-}
-
-CMutableTransaction CreateNewContextualCMutableTransaction(const Consensus::Params& consensusParams, int nHeight, int nExpiryDelta) {
     CMutableTransaction mtx;
-    bool isOverwintered = NetworkUpgradeActive(nHeight, consensusParams, Consensus::UPGRADE_OVERWINTER);
+    bool isOverwintered = consensusParams.NetworkUpgradeActive(nHeight, Consensus::UPGRADE_OVERWINTER);
     if (isOverwintered) {
         mtx.fOverwintered = true;
-        mtx.nExpiryHeight = nHeight + nExpiryDelta;
+        mtx.nExpiryHeight = nHeight + (expiryDeltaArg ? expiryDeltaArg.get() : consensusParams.DefaultExpiryDelta(nHeight));
 
+        // Review: Should the following if() be checking the expiry delta rather than the transaction height?
```

Comment:
No. TX_EXPIRY_HEIGHT_THRESHOLD is an absolute height. (It's the height at which the semantics of locktime changes, so we decided that expiry heights should be less than that.)